superset/connectors/sqla/models.py

```diff
@@ -1427,7 +1427,20 @@ def _get_top_groups(  # pylint: disable=no-self-use
 
         return or_(*groups)
 
+    def get_filter_type(self, filter: dict) -> str:
+        for column in self.columns:
+            if filter['col'] == column.column_name:
+                return column.type
+
     def query(self, query_obj: QueryObjectDict) -> QueryResult:
+        # Converts comparator from string to boolean if field type is boolean
+        for i in range(len(query_obj['filter'])):
+            if self.get_filter_type(query_obj['filter'][i]) == 'BOOLEAN':
```

Comment:
Could this whole thing not be avoided if we just have 1439, because we are doing a lot of iterating. Because if the value is true, we can presume that its a boolean.